src/platform/windows.rs

```diff
@@ -51,6 +51,51 @@ pub trait EventLoopBuilderExtWindows {
     /// # }
     /// ```
     fn with_dpi_aware(&mut self, dpi_aware: bool) -> &mut Self;
+
+    /// A callback, executed before dispatching a win32 message to the window procedure, where you
+    /// can call [`TranslateAcceleratorW`](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-translateacceleratorw) to
+    /// processes accelerator keys for menu commands.
+    ///
+    /// The callback return value is used as an indicator to whether the message
+    /// should be translated using [`TranslateMessage`](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-translatemessage)
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// # use winapi::um::winuser::{ACCEL, VkKeyScanW, VK_CONTROL, TranslateAcceleratorW, CreateAcceleratorTableW};
+    /// # use crate::winit::{event_loop::EventLoopBuilder, platform::windows::EventLoopBuilderExtWindows};
+    /// thread_local! {
+    ///     static ACCELS:Vec<ACCEL> = {
+    ///         unsafe {
+    ///             vec![ACCEL {
+    ///                 cmd: 554,
+    ///                 key: VkKeyScanW('a' as u16) as _  ,
+    ///                 fVirt: VK_CONTROL as _,
+    ///             }]
+    ///         }
+    ///     }
+    /// }
+    ///
+    /// fn main() {
+    ///     let event_loop = EventLoopBuilder::new()
+    ///         .with_translate_accel_callback(Box::new(|mut m| unsafe {
+    ///             let mut translated = false;
+    ///             ACCELS.with(|accels| {
+    ///                 translated = TranslateAcceleratorW(
+    ///                     m.hwnd,
+    ///                     CreateAcceleratorTableW(accels.as_ptr() as _, 1),
+    ///                     &mut m,
+    ///                 ) == 1;
+    ///             });
+    ///             translated
+    ///         }))
+    ///         .build();
+    /// }
+    /// ```
+    fn with_translate_accel_callback(
```

Comment:
As the implementation is not tied to to accelerators internally, we could also just give it a more general name like `with_msg_hook`

Reply:
I like that, this could also let users disable winit's internal Dispatch/Translate and allow them to use their own calls to Dispatch/Translate.